From 797c56efbb8aa8653549a4e4a38d4c9f2ceace28 Mon Sep 17 00:00:00 2001
From: Stefan Schmidt <stefan.schmidt@huawei.com>
Date: Fri, 21 May 2021 22:44:31 +0200
Subject: [PATCH] util/netif-mgmt: only include missing strlcpy header if
 actually missing

In commit 4e2ef143046a0c89afe6fd66acf5b7a7f35e8356 this header was
added unconditionally. For my system (Yocto with musl) this results in
strlcpy being redefined to __missing_strlcpy in all cases. No matter if
strlcpy is available or not.

Given it is available and the nl.m4 macros detects this, the libstrlcpy.la
helper is not built which results in:

netif-mgmt.c:386: more undefined references to `___missing_strlcpy' follow

Upstream-Status: Submitted [https://github.com/openthread/wpantund/pull/497]

Signed-off-by: Stefan Schmidt <stefan.schmidt@huawei.com>
---
 src/util/netif-mgmt.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/util/netif-mgmt.c b/src/util/netif-mgmt.c
index 3fffb2e..8fa0107 100644
--- a/src/util/netif-mgmt.c
+++ b/src/util/netif-mgmt.c
@@ -69,7 +69,9 @@
 #define IFEF_NOAUTOIPV6LL   0x2000  /* Interface IPv6 LinkLocal address not provided by kernel */
 #endif
 
+#ifdef MISSING_STRLCPY
 #include "missing/strlcpy/strlcpy.h"
+#endif
 
 #ifndef SIOCSIFLLADDR
 #define SIOCSIFLLADDR SIOCSIFHWADDR
